### PR TITLE
Pass resources to pooler

### PR DESCRIPTION
**Pull request by ianstanton**
- Allow for applying resources to connection pooler
- Change `resources` type from `ResourceRequirements` to `PoolerTemplateSpecContainersResources`
- Pass `resources` all the way down to pooler spec

### CODE REVIEW

File: tembo-operator/yaml/sample-with-pooler.yaml [hunk 1]
```diff
@@ -22,12 +22,18 @@ spec:
...
+  connectionPooler:
     enabled: true
     pooler:
       poolMode: transaction
       # Valid parameter values can be found at https://www.pgbouncer.org/config.html
       parameters:
         default_pool_size: "50"
         max_client_conn: "5000"
-      resources: ~
+      resources:
```
Comment (ianstanton) [inline tembo-operator/yaml/sample-with-pooler.yaml:33]: Add example to sample yaml

File: tembo-operator/src/defaults.rs [hunk 2]
```diff
@@ -133,9 +133,13 @@ pub fn default_conn_pooler_enabled() -> bool {
...
+pub fn default_pool_mode() -> PoolerPgbouncerPoolMode {
+    PoolerPgbouncerPoolMode::Transaction
+}
+
 pub fn default_pgbouncer() -> PgBouncer {
     PgBouncer {
-        poolMode: PoolerPgbouncerPoolMode::Transaction,
+        poolMode: default_pool_mode(),
         parameters: None,
         // TODO(ianstanton): Add some sensible defaults for pgbouncer resources
```
Comment (ianstanton) [inline tembo-operator/src/defaults.rs:144]: Will complete this TODO before merging

File: tembo-operator/src/apis/coredb_types.rs [hunk 3]
```diff
@@ -115,10 +115,11 @@ pub struct ConnectionPooler {
 #[derive(Deserialize, Serialize, Clone, Debug, JsonSchema, Default)]
 #[allow(non_snake_case)]
 pub struct PgBouncer {
+    #[serde(default = "defaults::default_pool_mode")]
     pub poolMode: PoolerPgbouncerPoolMode,
     // Valid parameter values can be found at https://www.pgbouncer.org/config.html
     pub parameters: Option<BTreeMap<String, String>>,
-    pub resources: Option<ResourceRequirements>,
+    pub resources: Option<PoolerTemplateSpecContainersResources>,
```
Comment (ianstanton) [inline tembo-operator/src/apis/coredb_types.rs:122]: Necessary change in order to pass to pgbouncer template in cnpg.rs

File: tembo-operator/src/defaults.rs [hunk 4]
```diff
@@ -133,9 +133,13 @@ pub fn default_conn_pooler_enabled() -> bool {
     false
 }
 
+pub fn default_pool_mode() -> PoolerPgbouncerPoolMode {
+    PoolerPgbouncerPoolMode::Transaction
```
Comment (ianstanton) [inline tembo-operator/src/defaults.rs:137]: Move default to fn

**Review by sjmiller609 — APPROVED** [review]

**Pull request merged**